app/waf/src/controllers/adc.controller.ts

```diff
@@ -342,11 +356,27 @@ export class AdcController extends BaseController {
       tenantId: await this.tenantId,
     });
 
-    if (adc.type === 'HW' && !(await this.untrustAdc(adc))) {
-      throw new HttpErrors.UnprocessableEntity('Fail to untrust device');
+    if (adc.type === 'HW') {
```

Comment:
For the hw branch, we may need to modify further? If yes, could we just skip the change this time?

Reply:
HW branch is kept as it is now except adding the `await this.adcRepository.deleteById(id);` to keep it in the same behavior with VE. 
(You may notice the combined conditions divided into 2 here, however, they are actually in the same logic.)